pages/game.tsx

```diff
@@ -10,8 +10,25 @@ import DrawPile from "../components/gameboard/DrawPile";
 import DiscardPile from "../components/gameboard/DiscardPile";
 import CardGrid from "../components/gameboard/CardGrid";
 import OpponentCardGrid from "../components/gameboard/OpponentCardGrid";
+import { SocketContext } from "../contexts/SocketContext";
+import { useContext, useEffect } from "react";
+import { getPlayerName, getSocketID } from "../lib/functions";
 
 export default function Game() {
+  const { socket } = useContext(SocketContext);
+  const playerName = getPlayerName();
+  const socketID = getSocketID();
```

Comment:
Not a next.js expert, but I think this is the reason your build fails, take a look at the output

<img width="733" alt="Screenshot 2021-03-28 at 12 50 08" src="https://user-images.githubusercontent.com/48794499/112749831-1cd54400-8fc5-11eb-95cc-d2ae0c418e5d.png">

Next tries to prerender this, but `localStorage` is a browser api..

I think keeping `playerName` and `socketID` in a state and setting it in the useEffect hook will fix this, not sure..

Aside from that, what happens when the items are `null`?

Reply:
Thanks for the suggestions!
@jamarob so I tried setting playerName and socketID as states and set them in the useEffect. This causes the server to crash when connecting to a game 😨 

I removed the state again and put ` const playerName = getPlayerName();
    const socketID = getSocketID();` inside the useEffect - maybe this will actually fix the localStorage error.
I'll push this in a bit and check if deployment etc. will succeed.

Reply:
Ahh, you only need them inside the useEffect so the states are not necessary, I guess. Sorry, can't say why this crashes 🤷

Reply:
Deployment works now! Thanks again :)